packages/core/src/components/select/select.tsx

```diff
@@ -40,18 +41,54 @@ export class AtomSelect {
     label?: string
     selected?: boolean
     disabled?: boolean
+    badge?: { type: string; label: string }
   }> = []
 
   @Event() atomBlur!: EventEmitter<void>
   @Event() atomCancel!: EventEmitter<void>
   @Event() atomChange!: EventEmitter<string>
   @Event() atomDismiss!: EventEmitter<void>
   @Event() atomFocus!: EventEmitter<void>
+  @Method() setBadge() {
+    const ionItemElements = document.querySelectorAll('ion-item')
 
+    ionItemElements.forEach((itemElement) => {
+      const elementText = itemElement.textContent
+      const optionWithBadge = this.optionsWhitBadge[elementText]
+
+      if (optionWithBadge) {
+        const badgeElement = document.createElement('atom-badge')
+        const { type, label } = optionWithBadge.badge
+
+        badgeElement.setAttribute('type', type)
+        badgeElement.textContent = label
+
+        itemElement.style.width = 'fit-content'
+        itemElement.appendChild(badgeElement)
+      }
+    })
+  }
+
+  filterOptionsWithBadge = (options) => {
+    return options?.reduce((optionsWhitBadge, option) => {
+      if (option?.badge?.label) {
+        const label = option.label || option.value
+
+        optionsWhitBadge[label] = option
+      }
+
+      return optionsWhitBadge
+    }, {})
+  }
```

Comment:
block: we have space to improve this code, we also need to avoid execution errors and is necessary to add a class inside the badge to customize it

```suggestion
  
  @Method()
  setBadge() {
    const ionItemElements = document.querySelectorAll('ion-item')
    const optionsWithBadge = this.optionsWithBadge

    if (!ionItemElements || !optionsWithBadge)

    ionItemElements?.forEach((itemElement) => {
      const elementText = itemElement.textContent?.trim()
      const optionWithBadge = optionsWithBadge[elementText]

      if (!optionWithBadge && !itemElement) return

      const badgeElement = document.createElement('atom-badge')
      const { type, label } = optionWithBadge.badge;

      badgeElement.setAttribute('type', type)
      badgeElement.textContent = label
      badgeElement.classList.add('atom-badge')

      itemElement.style.width = 'fit-content'
      itemElement.appendChild(badgeElement)
    })
  }

  filterOptionsWithBadge = (options: Array<{ label?: string; value?: string; badge?: { label: string; type: string } }>) => {
    return options?.reduce((optionsWithBadge, option) => {
      if (option?.badge?.label) {
        const label = option.label || option.value

        if (label) {
          optionsWithBadge[label] = option
        }
      }
      return optionsWithBadge
    }, {})
  }
```

Reply:
done https://github.com/juntossomosmais/atomium/pull/549/commits/06a7f0a30ca64c3be5b229e69e854942d056dad3